src/datatype/svecf32.rs

```diff
@@ -0,0 +1,770 @@
+        }
+    }
+    static SBSROUTINES: pgrx::pg_sys::SubscriptRoutines = pgrx::pg_sys::SubscriptRoutines {
+        transform: Some(transform),
+        exec_setup: Some(exec_setup),
+        fetch_strict: true,
+        fetch_leakproof: false,
+        store_leakproof: false,
+    };
+    std::ptr::addr_of!(SBSROUTINES).into()
+}
+
+#[cfg(not(any(feature = "pg14", feature = "pg15", feature = "pg16")))]
+#[pgrx::pg_extern(sql = "\
+CREATE FUNCTION _vectors_svecf32_subscript(internal) RETURNS internal
+IMMUTABLE STRICT PARALLEL SAFE LANGUAGE c AS 'MODULE_PATHNAME', '@FUNCTION_NAME@';")]
+fn _vectors_svecf32_subscript(_fcinfo: pgrx::pg_sys::FunctionCallInfo) -> Datum {
+    unreachable!()
+}
+
+#[pgrx::pg_extern(sql = "\
+CREATE FUNCTION _vectors_svecf32_send(svector) RETURNS bytea
+IMMUTABLE STRICT PARALLEL SAFE LANGUAGE c AS 'MODULE_PATHNAME', '@FUNCTION_NAME@';")]
+fn _vectors_svecf32_send(vector: SVecf32Input<'_>) -> Datum {
+    use pgrx::pg_sys::StringInfoData;
+    unsafe {
+        let mut buf = StringInfoData::default();
+        let dims = vector.dims;
+        let len = vector.len;
+        let data = vector.data();
+        pgrx::pg_sys::pq_begintypsend(&mut buf);
+        pgrx::pg_sys::pq_sendbytes(&mut buf, (&dims) as *const u16 as _, 2);
+        pgrx::pg_sys::pq_sendbytes(&mut buf, (&len) as *const u16 as _, 2);
+        pgrx::pg_sys::pq_sendbytes(
+            &mut buf,
+            data.indexes.as_ptr() as _,
+            (std::mem::size_of::<u16>() * len as usize) as _,
+        );
+        pgrx::pg_sys::pq_sendbytes(
+            &mut buf,
+            data.values.as_ptr() as _,
+            (std::mem::size_of::<F32>() * len as usize) as _,
+        );
+        Datum::from(pgrx::pg_sys::pq_endtypsend(&mut buf))
+    }
+}
+
+#[pgrx::pg_extern(sql = "
+CREATE FUNCTION _vectors_svecf32_recv(internal, oid, integer) RETURNS svector
+IMMUTABLE STRICT PARALLEL SAFE LANGUAGE c AS 'MODULE_PATHNAME', '@FUNCTION_NAME@';")]
+fn _vectors_svecf32_recv(internal: pgrx::Internal, _oid: Oid, _typmod: i32) -> SVecf32Output {
+    use pgrx::pg_sys::StringInfo;
+    unsafe {
+        let buf: StringInfo = internal.into_datum().unwrap().cast_mut_ptr();
+        let dims = (pgrx::pg_sys::pq_getmsgbytes(buf, 2) as *const u16).read_unaligned();
+        let len = (pgrx::pg_sys::pq_getmsgbytes(buf, 2) as *const u16).read_unaligned();
+        if dims == 0 || len == 0 {
+            pgrx::error!("data corruption is detected");
+        }
+        let indexes_bytes = std::mem::size_of::<u16>() * len as usize;
+        let indexes_ptr = pgrx::pg_sys::pq_getmsgbytes(buf, indexes_bytes as _);
+        let values_bytes = std::mem::size_of::<F32>() * len as usize;
+        let values_ptr = pgrx::pg_sys::pq_getmsgbytes(buf, values_bytes as _);
+        let mut output = SVecf32::new_zeroed_in_postgres(len as usize);
+        output.dims = dims;
+        std::ptr::copy(indexes_ptr, output.indexes_mut() as _, indexes_bytes);
```

Comment:
Let's detect possible data corruption for indexes.

Reply:
done in new commit